executor/ddl.go

```diff
@@ -224,6 +238,42 @@ func (e *DDLExec) executeTruncateTable(s *ast.TruncateTableStmt) error {
 	return err
 }
 
+func (e *DDLExec) executeTruncateLocalTemporaryTable(s *ast.TruncateTableStmt) error {
+	dbInfo, exists := e.ctx.GetInfoSchema().(infoschema.InfoSchema).SchemaByName(s.Table.Schema)
+	if !exists {
+		return infoschema.ErrDatabaseNotExists.GenWithStackByArgs(s.Table.Schema)
+	}
+
+	localTempTables := e.getLocalTemporaryTables()
+	if localTempTables == nil {
+		return infoschema.ErrTableNotExists.GenWithStackByArgs(s.Table.Schema, s.Table.Name)
+	}
+
+	tbl, exists := localTempTables.TableByName(s.Table.Schema, s.Table.Name)
```

Comment:
When there are two ways to do one thing, it get stuck.

Why not use `dbInfo.TableByName()` ?

Reply:
TableByName will first query localTempTables and if it is not found, it will next the original infoschema. The second step is unnecessary. But I agree with you it's better to query it from infoschema, maintainability is more important here.